src/main/java/org/pentaho/metadata/repository/IMetadataDomainRepositoryDatasourceType.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2022 Hitachi Vantara..  All rights reserved.
+ */
+
+package org.pentaho.metadata.repository;
+
+import java.util.Set;
+
+/**
+ * This interface defines a metadata domain repository, used to maintain a system-wide set of metadata domains.
+ * Includes the ability to retrieve Domains based on data source type.
+ */
+public interface IMetadataDomainRepositoryDatasourceType extends IMetadataDomainRepository {
```

Comment:
When extending types, whether classes or interfaces, I try to preserve the name suffix that indicates the type kind/pattern. In this case the suffix would be `Repository`, and, more specifically, `MetadataDomain~Repository` (a `Repository` _of_ `MetadataDomain`).

This makes it easy for me to identify the class kind/pattern (and any required "arguments" of it) by looking at the type name's suffix.

To be able to do this, depending on the involved English grammar, it's frequently needed to resort to some "connective". For example, for this case, I can think of `IDataSourceType~Aware~MetadataDomain~Repository`, or, more questionably, `IMetadataDomain~With~DataSourceType~Repository`.

On a related case, for implementations (of interfaces), this package uses `FileBased~MetadataDomain~Repository` and `InMemory~MetadataDomain~Repository`.

What do you think?

Reply:
You are correct, it should go before like `List` and `ArrayList`. But based on your suggestion and the team's agreement, we might not have to create a new interface.

Reply:
You are correct, it should go before like `List` and `ArrayList`. But based on your suggestion and the team's agreement, we might not have to create a new interface.